src/applications/vaos/new-appointment/components/CommunityCareProviderSelectionPage/ProviderSelectionField.jsx

```diff
@@ -1,5 +1,236 @@
+const INITIAL_PROVIDER_DISPLAY_COUNT = 5;
+
+function ProviderSelectionField({
+  address,
+  formData,
+  onChange,
+  idSchema,
+  requestStatus,
+  requestProvidersList,
+  communityCareProviderList,
+}) {
+  const [checkedProvider, setCheckedProvider] = useState();
+  const [mounted, setMounted] = useState(false);
+  const [showProvidersList, setShowProvidersList] = useState(false);
+  const [providersListLength, setProvidersListLength] = useState(
+    INITIAL_PROVIDER_DISPLAY_COUNT,
+  );
+  const currentlyShownProvidersList = communityCareProviderList?.slice(
+    0,
+    providersListLength,
+  );
+  const loadingProviders =
+    requestStatus === FETCH_STATUS.loading ||
+    requestStatus === FETCH_STATUS.notStarted;
+  const providerSelected = 'id' in formData;
+  useEffect(() => {
+    requestProvidersList(address);
+    setMounted(true);
+  }, []);
+
+  useEffect(
+    () => {
+      if (showProvidersList && !loadingProviders) {
+        scrollAndFocus('h2');
+      } else if (mounted && !loadingProviders) {
+        scrollAndFocus('.va-button-link');
+      }
+    },
+    [showProvidersList],
+  );
+
+  return (
+    <div className="vads-u-background-color--gray-lightest small-screen:vads-u-padding--2 medium-screen:vads-u-padding--3">
+      {!showProvidersList &&
+        !providerSelected && (
+          <button
+            className="va-button-link"
+            type="button"
+            onClick={() => setShowProvidersList(true)}
+          >
+            <i className="fas fa-plus vads-u-padding-right--0p5" />
+            Choose a provider
+          </button>
+        )}
+      {!showProvidersList &&
+        providerSelected && (
+          <>
+            <span className="vads-u-display--block vads-u-font-weight--bold">
+              {formData?.name}
+            </span>
+            <span className="vads-u-display--block">
+              {formData?.address?.line}
+            </span>
+            <span className="vads-u-display--block">
+              {formData?.address?.city}, {formData?.address?.state}{' '}
+              {formData?.address?.postalCode}
+            </span>
+            <span className="vads-u-display--block vads-u-font-size--sm vads-u-font-weight--bold">
+              {distanceBetween(
+                formData?.position?.latitude,
+                formData?.position?.longitude,
+                address.latitude,
+                address.longitude,
+              )}{' '}
+              miles
+            </span>
+            <div className="vads-u-display--flex">
+              <button
+                type="button"
+                className="vaos-appts__cancel-btn va-button-link vads-u-margin--0 vads-u-flex--0 vads-u-margin-right--2"
+                onClick={() => {
+                  setProvidersListLength(INITIAL_PROVIDER_DISPLAY_COUNT);
+                  setShowProvidersList(true);
+                }}
+              >
+                Change Provider
```

Comment:
Should be sentence case ("Change provider")

Reply:
Fixed